robot_skills/src/robot_skills/robot.py

```diff
@@ -195,6 +329,36 @@ def __exit__(self, exception_type, exception_val, trace):
             rospy.logerr("Robot exited with {0},{1},{2}".format(exception_type, exception_val, trace))
         self.close()
 
+def _collect_needs_desires(needs, desires, test_func):
```

Comment:
Check whitespaces etc. with decent linter

Reply:
They are functions, rather than methods, but not so sure. While the code itself is generic, how often do you look for matching a need?

However, it can be moved of course. I haven't run into any generic common set of python code yet (eg also for the decorators).

Reply:
I think its fine to leave these here, as they are only used here and _marked as private/protected.